src/validate/mod.rs

```diff
@@ -0,0 +1,364 @@
+use super::parse::*;
+
+pub fn validate(schemas: Vec<Schema>) -> Vec<Schema> {
+    let mut validated: Vec<Schema> = Vec::new();
+
+    for schema in schemas {
+        let validated_schema = match validated.iter_mut().find(|v| v.name == schema.name) {
+            Some(v) => v,
+            None => {
+                validated.push(Schema::new(schema.name));
+                validated.last_mut().unwrap()
+            }
+        };
```

Comment:
If someone defines the same schema twice, should this actually continue to 'dedupe' them and combine tables/records into a single `Schema` struct? And likewise for a table that's declared in multiple locations? It helps when validating that multiple named records aren't defined for the same table, but it's not the only way to accomplish that.